x-pack/plugins/lens/server/embeddable/make_lens_embeddable_factory.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EmbeddableRegistryDefinition } from 'src/plugins/embeddable/server';
+import type { SerializableRecord } from '@kbn/utility-types';
+import {
+  mergeMigrationFunctionMaps,
+  MigrateFunctionsObject,
+} from '../../../../../src/plugins/kibana_utils/common';
+import { DOC_TYPE } from '../../common';
+import {
+  commonMakeReversePaletteAsCustom,
+  commonRemoveTimezoneDateHistogramParam,
+  commonRenameFilterReferences,
+  commonRenameOperationsForFormula,
+  commonUpdateVisLayerType,
+} from '../migrations/common_migrations';
+import {
+  LensDocShape713,
+  LensDocShape715,
+  LensDocShapePre712,
+  VisState716,
+  VisStatePre715,
+} from '../migrations/types';
+import { extract, inject } from '../../common/embeddable_factory';
+
+export const makeLensEmbeddableFactory =
+  (filterMigrations: MigrateFunctionsObject) => (): EmbeddableRegistryDefinition => {
+    return {
+      id: DOC_TYPE,
+      migrations: mergeMigrationFunctionMaps(filterMigrations, {
```

Comment:
This doesn't work right - it's trying to apply the `filterMigrations` to the whole embedded Lens vis instead of each specific filter. I guess `filterMigrations` needs to be put through `getLensFilterMigrations` first.

Reply:
We can add a unit test for this by passing a dummy migration like the one I mentioned into `makeLensEmbeddableFactory`, then check whether `makeLensEmbeddableFactory(dummyMigration)()?.migrations[8.1](testEmbeddedSavedObject)` is applying it correctly.